quiche/src/lib.rs

```diff
@@ -6816,11 +6816,26 @@ impl Connection {
             .as_ref()
             .map_or(false, |conn_err| !conn_err.is_app)
         {
+            let mut force_downgrade = false;
             let epoch = match self.handshake.write_level() {
                 crypto::Level::Initial => packet::Epoch::Initial,
                 crypto::Level::ZeroRTT => unreachable!(),
                 crypto::Level::Handshake => packet::Epoch::Handshake,
-                crypto::Level::OneRTT => packet::Epoch::Application,
+                crypto::Level::OneRTT => {
+                    if self.handshake_confirmed {
+                        packet::Epoch::Application
+                    } else {
+                        // If the handshake hasn't been confirmed, we can't send
+                        // 1-RTT packets since the peer
+                        // may not have the correct keys. Downgrade the status to
+                        // Handshake - if required, we'll downgrade further to
+                        // Initial later on.
+                        //
+                        // https://datatracker.ietf.org/doc/html/rfc9000#name-immediate-close-during-the-
+                        force_downgrade = true;
```

Comment:
The problem with checking `handshake_confirmed` naively is that the client drops the Initial epoch state upon sending a Handshake packet. That means that downgrading all `Handshake` epoch packets naively could cause the client to drop the newly-downgraded packet

This boolean is super ugly, so I'm very open to changing it if that's preferre. I see two other alternatives:
1. Refactor such that we can trigger another CC frame getting flushed in a separate packet space. Unless I'm missing something (very possible), this seems like it'd be a lot of work. However, it's closer what the RFC recommends: 
> Prior to confirming the handshake, a peer might be unable to process 1-RTT packets, so an endpoint SHOULD send a CONNECTION_CLOSE frame in both Handshake and 1-RTT packets. A server SHOULD also send a CONNECTION_CLOSE frame in an Initial packet.
2. Change the client side so that we only drop the keys for the Initial space when the handshake gets confirmed, since that would force the client and server to be in sync.